performant-software/DM
# Text and viewer bug fixes

### What does this PR do?
- Addresses problems with character-escaping where '\' characters disappeared (strings of consecutive backslashes will still behave strangely) and '^' characters triggered a server-side error.
- Viewer containers now restore scroll position after a drag and drop to a new spot in the grid.
- Rollover window resize icon stays at the bottom right instead of scrolling upward with overflowing content. Delete icons no longer appear on resource summaries in the rollover window if the window is in the midst of resizing.
- Increases the max width of the project title / TOC button from 100px to 350px
- Adds the document icon to read-only clones so that annotations/targets may be viewed.
- Adds thumbnail image of canvas resource summaries when the resource as a whole has been linked (via document icon).

### What issues does it address?
Closes #119, closes #117, closes #124, closes #121, closes #109 

### How to test
- On [staging](http://67.207.158.85/), open a workspace.
- Try editing a text document to include \ and ^ characters, opening a new copy of the document, and reloading the page then reopening the document. Besides consecutive strings of backslashes losing characters upon reload, the text should appear as you entered it.
- Scroll a text document partway down and then drag and drop its container to a new spot in the grid. The scroll position should be the same as where you left it, not reset to the top. Try this with editable, clone, and editable-toggled-off documents.
- Open the rollover window for a target with several annotations/links and scroll through the list of resource summaries. The resize icon should stay put at the bottom right of the window. Drag it to start a resize and, while resizing, move the cursor over various resource summaries. The "delete" icon should not appear (it should still appear when resource summaries are hovered over if a resize is not in progress).
- Confirm that the TOC / project switcher button accommodates a reasonable length of text for the project title.
- Open a resource and a second instance of that resource as a read-only clone. For both, you should see a document icon in the upper right of its container, which will bring up the list of annotations/targets when clicked or hovered over. See note in #117 - the icon seems to have been hidden in read-only clones deliberately, but unclear why.
- Open an image resource and any other resource. Hover over the document icon in the upper right of the image resource's container and then click the link button in the rollover window. Select a target in another resource (e.g. its document icon). If you then hover over that target, you should see an annotation item for the first resource that includes a thumbnail of the image (previously, only the image's title would show).
